nifi-commons/nifi-hadoop-utils/src/main/java/org/apache/nifi/hadoop/KerberosProperties.java

```diff
@@ -0,0 +1,146 @@
+                            .build();
+                }
+
+                // Check that the Kerberos configuration is readable
+                if (!kerberosConfigFile.canRead()) {
+                    return new ValidationResult.Builder().subject(subject).input(input).valid(false)
+                            .explanation(String.format("unable to read Kerberos config [%s], please make sure the path is valid "
+                                    + "and nifi has adequate permissions", kerberosConfigFile.getAbsoluteFile()))
+                            .build();
+                }
+
+                return new ValidationResult.Builder().subject(subject).input(input).valid(true).build();
+            }
+        };
+
+        this.kerberosPrincipal = new PropertyDescriptor.Builder()
+                .name("Kerberos Principal")
+                .required(false)
+                .description("Kerberos principal to authenticate as. Requires nifi.kerberos.krb5.file to be set in your nifi.properties")
+                .addValidator(kerberosConfigValidator)
+                .build();
+
+        this.kerberosKeytab = new PropertyDescriptor.Builder()
+                .name("Kerberos Keytab").required(false)
+                .description("Kerberos keytab associated with the principal. Requires nifi.kerberos.krb5.file to be set in your nifi.properties")
+                .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+                .addValidator(kerberosConfigValidator)
+                .build();
+    }
+
+    public static KerberosProperties create(final NiFiProperties niFiProperties) {
+        if (niFiProperties == null) {
+            throw new IllegalArgumentException("NiFiProperties can not be null");
+        }
+        return new KerberosProperties(niFiProperties.getKerberosConfigurationFile());
+    }
+
+    public File getKerberosConfigFile() {
+        return kerberosConfigFile;
+    }
+
+    public Validator getKerberosConfigValidator() {
+        return kerberosConfigValidator;
+    }
+
+    public PropertyDescriptor getKerberosPrincipal() {
+        return kerberosPrincipal;
+    }
+
+    public PropertyDescriptor getKerberosKeytab() {
+        return kerberosKeytab;
+    }
+
+    public static List<ValidationResult> validatePrincipalAndKeytab(final String subject, final Configuration config, final String principal, final String keytab, final ComponentLog logger) {
+        final List<ValidationResult> results = new ArrayList<>();
+
+        // if security is enabled then the keytab and principal are required
+        final boolean isSecurityEnabled = SecurityUtil.isSecurityEnabled(config);
+
+        if (isSecurityEnabled && StringUtils.isBlank(principal)) {
+            results.add(new ValidationResult.Builder()
+                    .valid(false)
+                    .subject(subject)
+                    .explanation("Kerberos Principal must be provided when using a secure HBase")
```

Comment:
I'm guessing this is due to some refactoring, but we should probably not mention using a secure HBase in a general KerberosProperties class